net/v2ray-core/Makefile

```diff
@@ -0,0 +1,120 @@
+#
+# Copyright (C) 2019 Xingwang Liao
+# Copyright (C) 2019 Alonzo Liu <unl89@protonmail.com>
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=v2ray-core
+PKG_VERSION:=4.20.0
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/v2ray/v2ray-core/tar.gz/v$(PKG_VERSION)?
+PKG_HASH:=579a7098b8f6cc68c3f94c5453fa274299f8b072736f364654298bd4cf57d24e
+
+PKG_LICENSE:=MIT
+PKG_LICENSE_FILES:=LICENSE
+PKG_MAINTAINER:=Alonzo Liu <unl89@protonmail.com>
+
+PKG_BUILD_DEPENDS:=golang/host
+PKG_BUILD_PARALLEL:=1
+PKG_USE_MIPS16:=0
+
+GO_PKG:=v2ray.com/core
+
+include $(INCLUDE_DIR)/package.mk
+include ../../lang/golang/golang-package.mk
+
+define Package/v2ray-core/Default
+  TITLE:=A platform for building proxies
+  URL:=https://www.v2ray.com
+endef
+
+define Package/v2ray-core/Default/description
+  Project V is a set of network tools that help you to build your own computer network.
+  It secures your network connections and thus protects your privacy.
+endef
+
+define Package/v2ray-core
+  $(call Package/v2ray-core/Default)
+  TITLE+= (full)
+  DEPENDS:=$(GO_ARCH_DEPENDS) +ca-certificates
+endef
+
+define Package/v2ray-core/description
+$(call Package/v2ray-core/Default/description)
+
+  This package contains v2ray, v2ctl and v2ray-assets.
+endef
+
+GEOIP_VER:=20190516.1
+GEOIP_MD5:=b6f6b19e4028b8b7d569b424dadfc737
+GEOIP_FILE:=geoip.dat.$(GEOIP_VER)
+
+define Download/geoip
+  URL:=https://github.com/v2ray/geoip/releases/download/$(GEOIP_VER)/
+  URL_FILE:=geoip.dat
+  MD5SUM:=$(GEOIP_MD5)
```

Comment:
This should be changed to sha256 `HASH`.

Reply:
I suggest drop `GEOIP_MD5` and `GEOIP_FILE` and write hash here.
Because you can write `HASH:=skip` and then use:
```
make package/v2ray-core/download V=s
make package/v2ray-core/check FIXUP=1 V=s
```
To generate hash automatically. https://openwrt.org/docs/guide-developer/packages#use_source_repository

Reply:
The hashes can be set directly, but `GEOIP_FILE`/`GEOSITE_FILE` are also used in `Build/Prepare`.

Reply:
@unl89 Please change the `MD5SUM`s to (sha256) `HASH`s. (The wiki was outdated; I have updated the example code.)